src/hotspot/share/gc/shenandoah/shenandoahUtils.hpp

```diff
@@ -147,9 +147,15 @@ class ShenandoahSafepoint : public AllStatic {
   static inline bool is_at_shenandoah_safepoint() {
     if (!SafepointSynchronize::is_at_safepoint()) return false;
 
+    Thread* const thr = Thread::current();
+    // Shenandoah GC specific safepoints are scheduled by control thread,
+    // so that, querying from control thread can not happen during those
+    // safepoints.
```

Comment:
Consider this wording:

```
    // Shenandoah GC specific safepoints are scheduled by control thread.
    // So if we are enter here from control thread, then we are definitely not
    // at Shenandoah safepoint, but at something else.
```

Reply:
Updated accordingly.

Thanks, Aleksey.